Classes/Middleware/Captcha.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Blueways\BwCaptcha\Middleware;
+
+use Blueways\BwCaptcha\Utility\CaptchaBuilderUtility;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use TYPO3\CMS\Core\Routing\PageArguments;
+use TYPO3\CMS\Extbase\Configuration\ConfigurationManager;
+use TYPO3\CMS\Extbase\Configuration\ConfigurationManagerInterface;
+use TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication;
+use TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController;
+
+class Captcha implements MiddlewareInterface
+{
+    protected ResponseFactoryInterface $responseFactory;
+    protected ConfigurationManager $configurationManager;
+    protected TypoScriptFrontendController $tsfe;
+
+    public function __construct(
+        ResponseFactoryInterface $responseFactory,
+        ConfigurationManager $configurationManager,
+        TypoScriptFrontendController $tsfe
```

Comment:
In v12 the `TypoScriptFrontendController` cannot be loaded via DI anymore ([Changelog](https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-96107-DeprecatedFunctionalityRemoved.html)) - so this breaks the whole page.
Accessing the user session via globals `$GLOBALS['TSFE']->fe_user` still works.

Reply:
Thanks, good point. I will change this.

Reply:
Hey @dev-rke, how is it going? Looking forward to merge your PR :) 
Just let me know if you don't have time right know - I can also do these little adjustments.